spec/integration/outputs/kafka_spec.rb

```diff
@@ -191,6 +191,25 @@
     end
   end
 
+  context 'when setting message_headers' do
+    let(:num_events) { 10 }
+    let(:test_topic) { 'logstash_integration_topic4' }
+
+    before :each do
+      config = base_config.merge({"topic_id" => test_topic, "message_headers" => {"k1" => "v1"}})
```

Comment:
can you also add the case to validate the `event.sprintf`?

Reply:
Done